File: proyectoweb/src/main/java/co/edu/unicauca/competencias/proyectoweb/Teacher_module/Teacher_core/repositories/ITeacherRepository.ts
import { Teacher } from "../entities/Teacher";

export interface ITeacherRepository {
  findAll(): Promise<Teacher[]>;
  findById(id: number): Promise<Teacher | null>;
  create(Teacher: Teacher): Promise<Teacher>;
  update(id: number, Teacher: Teacher): Promise<Teacher | null>;
  delete(id: number): Promise<boolean>;
}

File: proyectoweb/src/main/java/co/edu/unicauca/competencias/proyectoweb/Teacher_module/Teacher_core/repositories/ITeacherRepository.java
package co.edu.unicauca.competencias.proyectoweb.Teacher_module.Teacher_core.repositories;
import co.edu.unicauca.competencias.proyectoweb.Teacher_module.Teacher_core.entities.Teacher;
import org.springframework.data.jpa.repository.JpaRepository;


public interface ITeacherRepository extends JpaRepository<Teacher, Integer> {
    Teacher findByNames(String names);
    boolean existsByNames(String names);
    boolean existsByIdentification(String identification);
}